app/gui/src/project-view/stores/suggestionDatabase/entry.ts

```diff
@@ -157,8 +157,8 @@ export function entryIsStatic(
 }
 
 /** Get the MethodPointer pointing to definition represented by the entry. */
-export function entryMethodPointer(entry: SuggestionEntry): MethodPointer | undefined {
-  if (entry.kind !== SuggestionKind.Method) return
+export function entryMethodPointer(entry: SuggestionEntry | undefined): MethodPointer | undefined {
```

Comment:
It seems to me we have a (loose) convention that responsibility for checking for a valid value belongs to the caller in such a case

Reply:
Allowing `undefined` inputs when the return can be `undefined` either way seems like a good thing though, because null coalescing operator doesn't help with nulls in argument positions, which makes calling those methods often annoying. External checks make more sense if the result is guaranteed for non-null values, because that way we prevent `undefined` types from showing up defensively in unexpected places.